### PR TITLE
basic structure setup with redux,axios and react-router-com

### PR DESCRIPTION
**Pull request by OsamaMalik651**

### CODE REVIEW

File: src/apis/weather.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import axios from "axios";
+
+const { REACT_APP_APP_ID } = process.env;
+const instance = axios.create({
+  baseURL: "https://api.openweathermap.org/data/2.5",
```
Comment (hafizmuhammadshoaib) [inline src/apis/weather.js]: base url should be inside .env file.

REACT_APP_BASE_URL=https://api.openweathermap.org/data/2.5
and call it here

`baseUrl: process.env.REACT_BASE_URL`

File: src/apis/weather.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  params: {
+    appid: REACT_APP_APP_ID,
+  },
+});
+
+// api.openweathermap.org/data/2.5/forecast?lat={lat}&lon={lon}&appid={6aa99e7ed05e0c81c4e743d589f8a7da}
```
Comment (hafizmuhammadshoaib) [inline src/apis/weather.js]: don't leave this type of comment in code

File: src/components/App.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import "../styles/App.css";
+import { BrowserRouter, Route, Routes } from "react-router-dom";
+import Home from "../containers/Home";
+
+function App() {
+  return (
+    <BrowserRouter>
+      <Routes>
+        <Route path="/" exact element={<Home />} />
```
Comment (hafizmuhammadshoaib) [inline src/components/App.js]: define constants folder in the project and create links.js file 

inside the constants folder, there should be 2 files 
- index.js (to export all the files of this folder from index.js file `export * from './links.js'`)
- links.js (inside links.js you will create `export const LINKS={home:{path:'/'}}`)

File: src/containers/Home.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  const [weatherData, setWeatherData] = useState({});
+  useEffect(() => {
+    setWeatherData(weather);
+  }, [weather]);
+  console.log(weatherData);
+  // function getDayWeather() {
+  //   const dayWeather = weatherData.list;
+  //   return dayWeather;
+  // }
```
Comment (hafizmuhammadshoaib) [inline src/containers/Home.js]: don't leave this type of comments

File: src/containers/Home.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  const getWeatherData = () => {
+    dispatch(getWeather());
+  };
+  return (
+    <div>
+      Home container
+      {/* {JSON.stringify(weatherData.list[1].dt_txt)} */}
```
Comment (hafizmuhammadshoaib) [inline src/containers/Home.js]: same as above

File: src/store/actions/weather.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import axios from "../../apis/weather";
+
+import * as actionTypes from "../action-types";
+
+export const getWeather = (lat, long) => async (dispatch) => {
+  //temporarily hardcoded for single city
+  // to be implemented: pass lat, long to get weather of searched city
+  const response = await axios.get("/forecast?lat=51.05&lon=-114.07");
```
Comment (hafizmuhammadshoaib) [inline src/store/actions/weather.js]: create `apiEndpoint.js` file constants folder like the way defined above

File: src/store/reducers/weatherReducer.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+const initialState = {
+  cod: "",
+  message: 0,
+  cnt: 0,
+  list: [],
+  city: {},
+};
```
Comment (hafizmuhammadshoaib) [inline src/store/reducers/weatherReducer.js:7]: +1 for creating an initial state :)

File: package.json [hunk 8]
```diff
@@ -6,9 +6,15 @@
     "@testing-library/jest-dom": "^5.16.2",
     "@testing-library/react": "^12.1.2",
     "@testing-library/user-event": "^13.5.0",
+    "axios": "^0.25.0",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
+    "react-redux": "^7.2.6",
+    "react-router-dom": "^6.2.1",
     "react-scripts": "5.0.0",
+    "reactstrap": "^9.0.1",
```
Comment (hafizmuhammadshoaib) [inline package.json:15]: please follow [this](https://reactstrap.github.io/?path=/story/home-installation--page) link to integrate react strap correctly

**Review by hafizmuhammadshoaib — CHANGES_REQUESTED** [review]

File: src/store/actions/weather.js (hunk 6 above)
Comment (hafizmuhammadshoaib) [reply src/store/actions/weather.js]: also, create a services folder and create a weather.js file. Inside the services/weather.js file create a method `getWeather`

File: src/apis/weather.js (hunk 1 above)
Comment (OsamaMalik651) [reply src/apis/weather.js]: Done

File: src/apis/weather.js (hunk 2 above)
Comment (OsamaMalik651) [reply src/apis/weather.js]: Done

File: src/components/App.js (hunk 3 above)
Comment (OsamaMalik651) [reply src/components/App.js]: Done

File: src/containers/Home.js (hunk 4 above)
Comment (OsamaMalik651) [reply src/containers/Home.js]: Done

File: src/containers/Home.js (hunk 5 above)
Comment (OsamaMalik651) [reply src/containers/Home.js]: Done

File: src/store/reducers/weatherReducer.js (hunk 7 above)
Comment (OsamaMalik651) [reply src/store/reducers/weatherReducer.js:7]: Thanks

File: package.json (hunk 8 above)
Comment (OsamaMalik651) [reply package.json:15]: Done

**Review by hafizmuhammadshoaib — APPROVED** [review]

**Pull request merged**